keps/sig-apps/2214-indexed-job/README.md

```diff
@@ -496,80 +499,99 @@ This feature has no node runtime implications.
 
 ### Rollout, Upgrade and Rollback Planning
 
-_This section must be completed when targeting beta graduation to a release._
+<!--
+This section must be completed when targeting beta to a release.
+-->
 
-* **How can a rollout fail? Can it impact already running workloads?**
+###### How can a rollout or rollback fail? Can it impact already running workloads?
 
   If the new kube-controller-manager crashes, it's possible that an older
   version of it would pick it up. In 1.21, when the IndexedJob feature is
   disabled (default), the controller would not sync Indexed Jobs, that is: the
   controller doesn't create or delete Pods and doesn't update Job status.
   Running Pods are not affected.
 
-* **What specific metrics should inform a rollback?**
+###### What specific metrics should inform a rollback?
 
   - job_sync_duration_seconds shows significantly more latency for label
     completion_mode=Indexed Jobs than completion_mode=NonIndexed.
   - job_sync_total shows more errors for completion_mode=Indexed than
     completion_mode=NonIndexed.
   - job_finished_total shows that Jobs with completion_mode=Indexed don't finish.
 
-* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
 
-  Manual test:
+  Manual test performed:
   
   1. Deploy k8s 1.21 cluster
   1. Upgrade to 1.22
   1. Create Indexed Job with big number of completions and pods that run for ~10min.
   1. Downgrade to 1.21. Verify that no new pods are created for the Indexed Job.
   1. Upgrade to 1.22. Verify that new pods are created for Indexed Job.
 
-* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
-fields of API types, flags, etc.?**
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
 
   No
 
 ### Monitoring Requirements
 
-_This section must be completed when targeting beta graduation to a release._
+<!--
+This section must be completed when targeting beta to a release.
+-->
 
-* **How can an operator determine if the feature is in use by workloads?**
+###### How can an operator determine if the feature is in use by workloads?
 
   - job_sync_total has values for the label completion_mode=Indexed.
 
-* **What are the SLIs (Service Level Indicators) an operator can use to determine 
-the health of the service?**
+###### How can someone using this feature know that it is working for their instance?
+
+- [x] Events
+  - Event Reason: SuccessfulCreate
+  - The message includes the pod name.
+- [x] API .status
+  - Condition name: 
+  - Other field: `completedIndexes` will not be empty as pods terminate.
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+  - per-day percentage of job_sync_total with label result=error <= 1%
+  - 99% percentile over day for job_sync_duration_seconds is <= 15s, assuming
```

Comment:
Was this measured on a cluster, it was ok for a beta feature but for a GA one I'd try go down with it, maybe?

Reply:
Let's see if we can get this metric in existing dashboards. @wojtek-t, could someone from SIG scalability assist?

Reply:
Can you please ping jkaniuk@ internally? He will delegate to someone.

Also - we re gathering metrics in scalability tests so we can check that, but as I mentioned on some other KEP update, this isn't super extensive usage of jobs. So I bet they will indeed be lowed. But that may not be strong enough signal to lower that. [Though conceptually I agree that 15s sounds pretty high for a single sync duration.]

Reply:
15s is assuming the creation of 500 pods though (the limit for a single sync). Is that high?

Reply:
> 15s is assuming the creation of 500 pods though (the limit for a single sync). Is that high?

Ohh yeah - I forgot about this. With that in mind it's acceptable.
I'm wondering if we can somehow distringuish that from the pure "in-controller' work (i.e. have some other metric or sth that will only be counting the "compute" part of the work (not the "execute on what I computed" one).

> So the 15s that I put here should be interpreted as: no change from existing performance.

Yeah - that makes sense. It's worth making that explicit maybe.

Reply:
Added the note.

I also added a GA requirement to have a scalability test in place.